opensrp-opd/src/main/java/org/smartregister/opd/OpdLibrary.java

```diff
@@ -224,6 +278,47 @@ public Event processOpdCheckInForm(@NonNull String eventType, String jsonString,
         opdCheckinEvent.addDetails(OpdConstants.Event.CheckIn.Detail.VISIT_ID, JsonFormUtils.generateRandomUUIDString());
         opdCheckinEvent.addDetails(OpdConstants.Event.CheckIn.Detail.VISIT_DATE, OpdUtils.convertDate(new Date(), OpdDbConstants.DATE_FORMAT));
 
-        return opdCheckinEvent;
+        return Collections.singletonList(opdCheckinEvent);
+    }
+
+    public List<Event> processOpdDiagnosisAndTreatmentForm(@NonNull String jsonString, @NonNull Intent data) throws JSONException {
+        JSONObject jsonFormObject = new JSONObject(jsonString);
+        JSONObject step1JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP1);
+        JSONObject step2JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP2);
+        JSONObject step3JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP3);
+        JSONObject step4JsonObject = jsonFormObject.optJSONObject(OpdConstants.JSON_FORM_EXTRA.STEP4);
+
+        String entityId = OpdUtils.getIntentValue(data, OpdConstants.IntentKey.BASE_ENTITY_ID);
+
+//        OpdCheckIn opdCheckIn = OpdLibrary.getInstance().getCheckInRepository().getLatestCheckIn(entityId);
+
+        String visitId = "visitId";
```

Comment:
Remove this hardcoded `visitId` and use the actual visit ID which is a UUID

Reply:
that was fixed

Reply:
👍